config/decoder.go

```diff
@@ -294,6 +295,41 @@ func (d *decoder) pointer(name string, value reflect.Value, def string) error {
 	return d.unmarshal(name, value.Elem(), def)
 }
 
+// Sets value to a channel type.
+func (d *decoder) channel(key string, value reflect.Value, def string) error {
+	return d.textUnmarshaller(key, value, def)
+}
+
+// Sets value to a function type.
+func (d *decoder) function(key string, value reflect.Value, def string) error {
+	return d.textUnmarshaller(key, value, def)
+}
+
+func (d *decoder) textUnmarshaller(key string, value reflect.Value, str string) error {
+	v := d.getGlobalProvider().Get(key)
+	if v.HasValue() {
+		str = v.String()
+	} else if str == "" {
+		return nil
+	}
+
+	if value.IsNil() {
+		value.Set(reflect.New(value.Type()).Elem())
+	}
+
+	// Value has to have a pointer receiver to be able to modify itself with TextUnmarshaller
+	if !value.CanAddr() {
+		return fmt.Errorf("can't use TextUnmarshaller because %s is not addressable", key)
```

Comment:
Prefer %q to %s to make it obvious which part is interpolated (it adds double quotes around the value)